bin/SymfonyPolyfillCleanup.php

```diff
@@ -0,0 +1,234 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Util;
+
+use Composer\Script\Event;
+
+/**
+ * Utilities to remove PHP 8.0 specific code of Symfony polyfills to make it compatible with woorelease.
+ * See issue: https://github.com/woocommerce/google-listings-and-ads/issues/331#issuecomment-838372148
+ *
+ * This class is based on https://github.com/woocommerce/google-listings-and-ads/blob/develop/bin/GoogleAdsCleanupServices.php
+ */
+class SymfonyPolyfillCleanup {
+
+	/**
+	 * Display debug output.
+	 *
+	 * @var boolean
+	 */
+	protected $debug = false;
+
+	/**
+	 * @var Event Composer event.
+	 */
+	protected $event = null;
+
+	/**
+	 * @var string The path of the Symfony library.
+	 */
+	protected $path = null;
+
+	/**
+	 * @var array The list of symfony polyfills.
+	 */
+	protected const LIST_OF_POLYFILLS = [
+		'polyfill-intl-normalizer',
+		'polyfill-mbstring',
+	];
+
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Event|null  $event Composer event.
+	 * @param string|null $path  Path of the Symfony library.
+	 */
+	public function __construct( Event $event = null, string $path = null ) {
+		$this->event = $event;
+		$this->path  = $path ?: dirname( __DIR__ ) . '/vendor/symfony/';
+	}
+
+	/**
+	 * Get list of Sympfony polyfills.
+	 *
+	 * @return array List of sympfony polyfills.
+	 */
+	protected function get_list_of_polyfills(): array {
+		return self::LIST_OF_POLYFILLS;
+	}
+
+	/**
+	 * Remove PHP 8.0 specific code of Symfony polyfills
+	 *
+	 * @param Event $event Event context provided by Composer
+	 */
+	public static function remove( Event $event = null ) {
+		$cleanup = new SymfonyPolyfillCleanup( $event );
+		$cleanup->remove_bootstraps80();
+
+	}
+
+	/**
+	 * Remove bootstrap80.php files and the conditional statement in bootstrap.php
+	 */
+	protected function remove_bootstraps80() {
+		$this->output_text( 'Removing PHP 8.0 specific code for symfony polyfills' );
+
+		$list_of_polyfills = $this->get_list_of_polyfills();
+
+		foreach ( $list_of_polyfills as $polyfill ) {
+
+			// Search for bootstrap.php
+			$bootstraps = array_unique(
+				$this->find_library_file_pattern( 'bootstrap', $polyfill )
+			);
+
+			foreach ( $bootstraps as $bootstrap ) {
+				$this->remove_statement_with_pattern( $bootstrap, "return require __DIR__.'/bootstrap80.php", 'if (\PHP_VERSION_ID >= 80000)' );
+			}
+
+			// Search for bootstrap80.php
+			$bootstraps80 = array_unique(
+				$this->find_library_file_pattern( 'bootstrap80', $polyfill )
+			);
+
+			foreach ( $bootstraps80 as $bootstrap80 ) {
+				$this->remove_file( $bootstrap80 );
```

Comment:
This will remove the file, even if the remove statement call fails. It's not an issue since it finds both, but should we prevent the file from being removed if the statement is not found?